### ISSUE TITLE
train mode and eval mode question 

### ISSUE THREAD

**Issue by shalijiang** [cornellius-gp/gpytorch#678]
I don't have much experience with pytorch, so just a very basic question about train mode and eval mode in gpytorch to make sure I understand the logic here.
say `model` is initialized with some `train_x, train_y`; if you call
`model.train()` 
`output=model(x)`
then `output` will be the prior joint distribution of y(x), and `train_x` and `train_y` were simply ignored in this computation; but if you call 
`model.eval()` 
`output=model(x)`
then `output` will be the posterior joint distribution of y(x), conditioned on `train_x` and `train_y`, correct? 

If my understanding is correct, could you point to me where was this implemented? I supposed there would be something like 
`if train_mode
       compute_prior
else
       condition on train and compute posterior
` in some `forward` function. 

Just curious, thanks!

**Comment by jacobrgardner** [comment]
Yup, that's correct! See here:

https://github.com/cornellius-gp/gpytorch/blob/747c96b09b42b5e4ee589ba152da2ac54ada0fe7/gpytorch/models/exact_gp.py#L195

**Comment by lagrassa** [comment]
What is the reasoning behind not using the training_inputs to compute the posterior in train() mode?  Isn't it useful to be able to train the GP hyperparameters using the training data (for the GP)?
